### PR TITLE
test: ensure that forking takes are received by all takers

### PR DESCRIPTION
**Pull request by forivall**
| Q                        | A  |
| ------------------------ | ---  |
| Fixed Issues?            | N/A |
| Patch: Bug Fix?          | No |
| Major: Breaking Change?  | No |
| Minor: New Feature?      | No |
| Tests Added + Pass?      | Yes |
| Any Dependency Changes?  | No |

I wanted to ensure that `take`s from multiple forked sagas are received by all of them for a single `put`. I didn't see anything in the documentation verifying this, and didn't see any tests verifying this scenario, so i added some tests to make sure that it's covered, and it is! Let me know if there are other tests for this. (my use case is indirectly composing sagas, ie, dispatching an action, and then waiting for the result action to dispatch/put)

I can also add a note in the documentation, probably in https://redux-saga.js.org/docs/advanced/FutureActions

### CODE REVIEW

**Comment by Andarist** [comment]
Thank you for your contribution - to merge this in I would have to dive into our test suite to verify if those tests are not duplicates of some existing ones. I might not be able to do that any time soon though because my plate is quite full right now. I hope you understand.

**Comment by forivall** [comment]
yeah, i understand. no worries.

File: packages/core/__tests__/interpreter/takeSync.js [hunk 1]
```diff
@@ -681,6 +681,35 @@ test('action dispatched in root saga should get scheduled and taken by a "siblin
...
 
+test('action dispatched in root saga should get scheduled and taken by 2 "siblings" takes', () => {
+  const reducer = (state, action) => {
+    if (!state) return []
+
+    return state.concat(action.type)
+  }
+
+  const middleware = sagaMiddleware()
+  const store = createStore(reducer, applyMiddleware(middleware))
```
Comment (neurosnap) [inline packages/core/__tests__/interpreter/takeSync.js:692]:
```suggestion
createStore(reducer, applyMiddleware(middleware))
```

File: packages/core/__tests__/interpreter/takeSync.js [hunk 2]
```diff
@@ -705,3 +734,31 @@ test('action dispatched synchronously in forked task should be taken a following
       expect(actual).toEqual([{ type: 'A', payload: 'foo' }])
     })
 })
+
+test('action dispatched synchronously in forked task should be taken a following sync take and forked take', () => {
+  const actual = []
+  const reducer = (state, action) => action.type
+
+  const middleware = sagaMiddleware()
+  const store = createStore(reducer, applyMiddleware(middleware))
```
Comment (neurosnap) [inline packages/core/__tests__/interpreter/takeSync.js:743]:
```suggestion
createStore(reducer, applyMiddleware(middleware))
```

**Review by neurosnap — CHANGES_REQUESTED** [review]
Thanks for your contribution and sorry for the delay.  I have reviewed all tests and don't see any tests that are similar to the ones written here.  It's possible that we are inadvertently testing these scenarios but I don't see anything wrong with including explicit tests written in this PR.

I made a couple of suggested changes (even though we are making the same mistake in previous tests).

**Comment by neurosnap** [comment]
I'm going to close this PR for now because it's been a year since it was actively worked on.  If someone wants to pick this up I'd be more than happy to review.  Thanks!

**Pull request closed without merging**